src/ucm/event/event.c

```diff
@@ -279,7 +279,7 @@ void *ucm_shmat(int shmid, const void *shmaddr, int shmflg)
     size_t size;
     int result;
 
-    ucm_event_enter();
+    ucm_event_enter_exclusive();
```

Comment:
we need ucm_event_enter_exclusive() only around the part which accesses the hash, but keep  ucm_event_dispatch(UCM_EVENT_SHMAT, &event) under non-exclusive lock

Reply:
release and lock again or guard hash with separate mutex?

Reply:
let's add separate pthread_spinlock_t (safest IMHO)
we can initialize it in UCS_STATIC_INIT{} in the end of the file